x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/json_diff.test.tsx

```diff
@@ -0,0 +1,290 @@
+    expect(removedLineBefore).toHaveStyle(`background: ${removedBackgroundColor}`);
+
+    /* LINE ADDITION */
+    /* Check that the added line is displayed only once */
+    expect(screen.queryAllByText(matchInOrder(['GPLv3']))).toHaveLength(1);
+
+    /* Added line appears in the "after" column */
+    const addedLine = getByText(matchInOrder(['+', 'license', 'GPLv3']));
+
+    const addedLineAfter = within(addedLine).getByText(matchInOrder(['license', 'GPLv3']));
+    expect(addedLineAfter).toHaveStyle(`background: ${addedBackgroundColor}`);
+  });
+
+  it('Rule actions and exception lists should not be shown as modified', () => {
+    const testAction = {
+      group: 'default',
+      id: 'my-action-id',
+      params: { body: '{"test": true}' },
+      action_type_id: '.webhook',
+      uuid: '1ef8f105-7d0d-434c-9ba1-2e053edddea8',
+      frequency: {
+        summary: true,
+        notifyWhen: 'onActiveAlert',
+        throttle: null,
+      },
+    } as const;
+
+    const testExceptionListItem = {
+      id: 'acbbbd86-7973-40a4-bc83-9e926c7f1e59',
+      list_id: '1e51e9b9-b7c0-4a11-8785-55f740b9938a',
+      type: 'rule_default',
+      namespace_type: 'single',
+    } as const;
+
+    const oldRule: RuleResponse = {
+      ...savedRuleMock,
+      version: 1,
+      actions: [testAction],
+    };
+
+    const newRule: RuleResponse = {
+      ...savedRuleMock,
+      version: 2,
+    };
+
+    /* Case: rule update doesn't have "actions" or "exception_list" properties */
+    const { rerender } = render(<RuleDiffTab oldRule={oldRule} newRule={newRule} />);
+    expect(screen.queryAllByText(matchInOrder(['actions']))).toHaveLength(0);
+
+    /* Case: rule update has "actions" and "exception_list" equal to empty arrays */
+    rerender(
+      <RuleDiffTab
+        oldRule={{ ...oldRule }}
+        newRule={{ ...newRule, actions: [], exceptions_list: [] }}
+      />
+    );
+    expect(screen.queryAllByText(matchInOrder(['actions']))).toHaveLength(0);
+
+    /* Case: rule update has an action and an exception list item */
+    rerender(
+      <RuleDiffTab
+        oldRule={{ ...oldRule }}
+        newRule={{
+          ...newRule,
+          actions: [{ ...testAction, id: 'my-other-action' }],
+          exceptions_list: [testExceptionListItem],
+        }}
+      />
+    );
+    expect(screen.queryAllByText(matchInOrder(['actions']))).toHaveLength(0);
+  });
+
+  it('Technical properties should not be included in preview', () => {
+    const oldRule: RuleResponse = {
+      ...savedRuleMock,
+      version: 1,
+      revision: 100,
+      created_at: '12/31/2023T23:59:999z',
+      created_by: 'mockUserOne',
+      updated_at: '01/01/2024T00:00:000z',
+      updated_by: 'mockUserTwo',
+    };
+
+    const newRule: RuleResponse = {
+      ...savedRuleMock,
+      version: 2,
+      revision: 1,
+      created_at: '12/31/2023T23:59:999z',
+      created_by: 'mockUserOne',
```

Comment:
Even though `created_at` and `created_by` **should never change**, let's change them here, in order to be 100% sure that they won't be displayed as diffs. Otherwise we are not actually testing that.

Reply:
Addressed in a follow-up commit